templates/common/_base/files/mtu-migration-dispatcher.yaml

```diff
@@ -0,0 +1,213 @@
+{{ if .Network }}{{ if .Network.MTUMigration }}{{ if .Network.MTUMigration.Machine }}{{ if .Network.MTUMigration.Machine.To -}}
+mode: 0755
+path: "/etc/NetworkManager/dispatcher.d/pre-up.d/99-mtu-migration.sh"
+contents:
+  inline: |
+    #!/bin/bash
+
+    set -ex
+
+    MTU_MIGRATION_DIR="/etc/cno/mtu-migration"
+    MTU_MIGRATION_CONFIG="${MTU_MIGRATION_DIR}/config"
+    MTU_MIGRATION_SAVE_DIR="${MTU_MIGRATION_DIR}/mtu"
+
+    if [ "$2" != "pre-up" ]; then
+        set +x
+        exit;
+    fi
+
+    if [ -z "$1" ]; then
+        echo "$0: called with no interface" 1>&2
+        set +x
+        exit 1;
+    fi
+
+    get_mtu_on_dev(){
+        local dev=$1
+        mtu_path="/sys/class/net/${dev}/mtu"
+        if [ ! -f "$mtu_path" ]; then
+          echo "ERROR: no mtu file found at $mtu_path" >&2
+          return 1
+        fi
+        cat $mtu_path
+    }
+
+    set_mtu_on_dev(){
+        local dev=$1
+        local c_type=$2
+        local new_mtu=$3
+        if [[ $c_type =~ .*ovs.* ]]; then
+            ovs-vsctl set int $dev mtu_request=$new_mtu
+        else
+            ip link set dev $dev mtu $new_mtu
+        fi
+    }
+
+    set_mtu_on_dev_routes() {
+        local dev=$1
+        local mtu=$2
+
+        dev_routes=$(ip route show dev $dev)
+
+        if [ -n "$dev_routes" ]; then
+            while IFS= read -r line; do
+                current_mtu=$(echo $line | sed -n -e 's/^.*mtu \([^ ]*\).*/\1/p')
+                echo "mtu=$current_mtu in this route: $line"
+                if [ -z "$current_mtu" ] || [ $current_mtu -gt $mtu ]; then
+                    ip route change ${line} dev $dev mtu ${mtu}
+                fi
+            done <<< "$dev_routes"
+        fi
+    }
+
+    save_mtu_for_dev(){
+        local dev=$1
+        local mtu=$2
+
+        if [ -f "${MTU_MIGRATION_SAVE_DIR}/${dev}" ]; then
+            rm -f "${MTU_MIGRATION_SAVE_DIR}/${dev}"
+        fi
+        if [ ! -d "${MTU_MIGRATION_SAVE_DIR}" ]; then
+            mkdir "${MTU_MIGRATION_SAVE_DIR}"
+        fi
+        echo "ORIGINAL_MTU=$mtu" > "${MTU_MIGRATION_SAVE_DIR}/${dev}"
+    }
+
+    set_mtu(){
+        local dev=$1
+        local c_type=$2
+        local is_vlan_parent=$3
+
+        if [ -f "${MTU_MIGRATION_SAVE_DIR}/${dev}" ]; then
+            . "${MTU_MIGRATION_SAVE_DIR}/${dev}"
+        else
+            ORIGINAL_MTU=$(get_mtu_on_dev "$dev")
```

Comment:
get_mtu_on_dev returns 1 if it isnt found and this logic seems to continue. Could that cause potential issues with assumign MTU is 1? Should you return -1 or something instead and exit?

Reply:
`get_mtu_on_dev` gives back the result of `cat ....`. The return is just the status of the function that should fail the script if different than 0 because of `set -x`. But let's change it because it is confusing. Just doing an an `exit 1` there should be equivalent.